datafusion/core/src/datasource/physical_plan/parquet/statistics.rs

```diff
@@ -766,10 +769,108 @@ impl<'a> StatisticsConverter<'a> {
         Ok(Arc::new(UInt64Array::from_iter(null_counts)))
     }
 
-    /// Returns a null array of data_type with one element per row group
-    fn make_null_array<I>(&self, data_type: &DataType, metadatas: I) -> ArrayRef
+    /// Extract the minimum values from Data page statistics
+    ///
+    /// In Parquet files, in addition to the Column Chunk level statistics
+    /// (stored for each column for each row group) there are also optional
+    /// statistics stored for each data page, part of the [Parquet Page Index].
+    /// Since a single Column Chunk is stored as one or more pages, page level statistics
+    /// can prune at a finer granularity.
+    ///
+    /// However since they are stored in a separate metadata structure
+    /// ([`Index`]) there is different code to extract them as arrow statistics
+    ///
+    /// Parameters:
+    ///
+    /// * `page_index`: The parquet page index, likely read from
+    /// [`ParquetMetadata::page_index()`]
+    ///
+    /// * row_group_indexes: The indexes of the row groups (indexes in
+    /// `page_index`) to extract the statistics from. This is an interator over `&usize` to
+    /// permit passing in  `&Vec<usize>` or similar
+    ///
+    /// # Return Value
+    ///
+    /// The returned array contains 1 value for each `NativeIndex` in the underlying
+    /// `Index`es, in the same order as they appear in `metadatas`.
+    ///
+    /// For example, if there are two `Index`es in `metadatas`:
+    /// 1. the first having `3` `PageIndex` entries
+    /// 2. the second having `2` `PageIndex` entries
+    ///
+    /// The returned array would have 5 rows
+    ///
+    /// Each value is either
+    /// * the minimum value for the page
+    /// * a null value, if the statistics can not be extracted
+    ///
+    /// Note that a null value does NOT mean the min value was actually
+    /// `null` it means it the requested statistic is unknown
+    ///
+    /// # Errors
+    ///
+    /// Reasons for not being able to extract the statistics include:
+    /// * the column is not present in the parquet file
+    /// * statistics for the pages are not present in the row group
+    /// * the stored statistic value can not be converted to the requested type
+    ///
+    /// # Example
+    /// ```no_run
+    /// tood
+    /// ```
+    pub fn data_page_mins<I>(
+        &self,
+        page_index: &ParquetColumnIndex,
+        row_group_indexes: I,
```

Comment:
I'll guess one reason why we want to pass in the `row_group_indexes` is due to the iteration over the row_group_indexes from the access_plan [here](https://github.com/apache/datafusion/blob/main/datafusion/core/src/datasource/physical_plan/parquet/page_filter.rs#L168).

We cannot assume we need all indices since access_plan does [filter ](https://github.com/apache/datafusion/blob/main/datafusion/core/src/datasource/physical_plan/parquet/access_plan.rs#L299-L307) based on `should_scan()` or not. 
Is this correct? If it is, then this was the missing piece in my prototype.

Reply:
> I'll guess one reason why we want to pass in the `row_group_indexes` is due to the iteration over the row_group_indexes from the access_plan [here](https://github.com/apache/datafusion/blob/main/datafusion/core/src/datasource/physical_plan/parquet/page_filter.rs#L168).
> 
> We cannot assume we need all indices since access_plan does [filter ](https://github.com/apache/datafusion/blob/main/datafusion/core/src/datasource/physical_plan/parquet/access_plan.rs#L299-L307) based on `should_scan()` or not. Is this correct? If it is, then this was the missing piece in my prototype.

Yes I think that is correct

I am still not super thrilled with this interface (mostly because it is different than `row_group_mins` etc that takes an interator directly.

However, I couldn't figure out how to make the types workout for making an iterator over Vec<Index> ...

Reply:
I had this interface in some version of my prototype, where it would take an iterator over all row_groups directly. However, this cannot be easily integrated with the existing code and the access_plan.row_group_indices. Perhaps, once the StatisticsConverter is fully used in `page_filter.rs` we can change the interface?